src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -113,6 +114,17 @@ private async Task ProcessOutgoingMessagesAsync(ServiceConnectionContext connect
                     var buffer = result.Buffer;
                     if (!buffer.IsEmpty)
                     {
+                        // We assume the first response message would be a HandshakeResponse.
+                        if (connection.IsMigrated)
+                        {
+                            if (SignalRProtocol.HandshakeProtocol.TryParseResponseMessage(ref buffer, out var message))
```

Comment:
If this is expected to be the first message for Migrated connection, move this parse logic to the top of the function outside of the while loop? Then there is no need to reset the `IsMigrated` property and this property can be readonly

Reply:
Sounds great.
I'll try implementing this and see what will happen.

Reply:
But make this kind of change could have unprecedented impacts on our existing logic so maybe we have to reconsider it carefully.